src/fragmentTypes.json

```diff
@@ -0,0 +1,30 @@
+{
```

Comment:
Could you explain the use of this file, please?

Reply:
so since at first we fetch for repository then after that for query, the data shape is different. If we didnt use this, only the field that is inside the first query will be saved and passed as props. and this warning appear `WARNING heuristic fragment matching going on`. so to enable the `updateMore` to merge the previous and more result including all its field, we need the IntrospectionFragmentMatcher to tell that there is 'union' of different data field